lib/mcrepo/debug_line_section.cpp

```diff
@@ -59,8 +59,8 @@ namespace pstore {
         std::uint8_t *
         debug_line_section_creation_dispatcher::write (std::uint8_t * const out) const {
             assert (this->aligned (out) == out);
-            auto * const scn =
-                new (out) debug_line_section (header_, section_->make_sources (), section_->align);
+            auto * const scn = new (out) debug_line_section (
+                header_digest_, header_, section_->make_sources (), section_->align);
```

Comment:
As you mentioned in [llvm-project-prepo@issues107](https://github.com/SNSystems/llvm-project-prepo/issues/107), the API change need to modify the compiler side as well. Will you change the compiler side? The pstore and llvm-project-prepo need to be merged back to the master branch at the same time. Otherwise, the compiler won't be built since the API change.

Reply:
I knew you are busy on the rld implementation, I am happy to solve the compiler issue107. Please let me know your thought.

Reply:
> Will you change the compiler side?

Yes. I'm working on it. 

This PR is only a draft ATM because committing this will break the compiler build until I've created a match PR there. Having another bad git day meant that the branch I created in the compiler has more changes in it than it should do.

Reply:
> Yes I'm working on it.

The compiler [PR is now live](https://github.com/SNSystems/llvm-project-prepo/pull/108) (as is this one).